From e65a2e739bea43948eb4bfcfcf10deb975f9fd73 Mon Sep 17 00:00:00 2001
From: zhengq <qi.zheng@intel.com>
Date: Thu, 7 May 2015 21:31:30 +0800
Subject: [PATCH] Touch: correct capability of touch input device.

Some APPs is depend on this capability, if don't match
the capability of touch chip, will make APPs don't work.

Change-Id: I5ea9cc5a64db6d6d549c0e85bbf63e6a8e8d621f
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-25010
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1574
Signed-off-by: zhengq <qi.zheng@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index 7a055830f586..76ded8c87673 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -2524,7 +2524,13 @@ static int mxt_initialize_t100_input_device(struct mxt_data *data)
 		goto err_free_mem;
 	}
 
-	input_set_abs_params(input_dev, ABS_MT_TOOL_TYPE, 0, MT_TOOL_MAX, 0, 0);
+	/* For T63 active stylus */
+	if (data->T63_reportid_min) {
+		input_set_capability(input_dev, EV_KEY, BTN_STYLUS);
+		input_set_capability(input_dev, EV_KEY, BTN_STYLUS2);
+		input_set_abs_params(input_dev, ABS_MT_TOOL_TYPE,
+			0, MT_TOOL_MAX, 0, 0);
+	}
 	input_set_abs_params(input_dev, ABS_MT_POSITION_X,
 			     0, data->max_x, 0, 0);
 	input_set_abs_params(input_dev, ABS_MT_POSITION_Y,
-- 
1.9.1

